src/silverlabnwb/metadata.py

```diff
@@ -35,25 +35,53 @@ def read_user_config():
     defaults.
     """
     with pkg_resources.resource_stream(__name__, 'metadata.yaml') as defaults:
-        settings = read_config_file(defaults)
+        settings, comments = read_base_config_file(defaults)
     if os.path.isfile(user_conf_path):
         with open(user_conf_path, 'r') as user_params:
-            settings = read_config_file(user_params, settings)
-    return settings
+            settings = update_config_file(user_params, settings)
+    return settings, comments
+
+
+def read_base_config_file(stream):
+    yaml_reader = yaml.YAML()
+    parsed_yaml_data = yaml_reader.load(stream)
+    yaml_data_comments = read_comments(parsed_yaml_data)
+    settings = {}
+    recursive_dict_update(settings, strip_strings(parsed_yaml_data))
+    return settings, yaml_data_comments
 
 
-def read_config_file(stream, base_settings=None):
+def read_comments(settings):
+    comments = {}
+    for k, v in settings.items():
+        if isinstance(v, collections.Mapping):
+            comments[k] = read_comments(v)
+        else:
+            comment = settings.ca.items.get(k, None)
+            if comment is not None:
+                yaml_comment = comment[2].value  # access CommentToken
+                clean_comment = ''
+                for line in yaml_comment.split('#'):
+                    stripped_line = line.strip()
+                    # ignore leading blank lines and similar useless info
+                    if stripped_line != '' and not stripped_line.startswith('>'):
+                        clean_comment = '\n'.join((clean_comment, stripped_line))
```

Comment:
If I understand right, we're building up the comment line by line. In that case, maybe it's better to do something like `clean_comment += f"\n{stripped_line}"` rather than joining repeatedly? That way the final string won't start with `\n`. I think `join` would be cleaner if we're passing it all the strings at once.